src/utilities.ts

```diff
@@ -152,93 +158,133 @@ export function getChecksumCommand() {
   return globalCheckSumCommand;
 }
 
-export function loadLaunchFile(jsonPath: string): any {
+export interface jsonDataType {
+  jsonData:any;
+  jsonDataAsString:string;
+}
+
+export function loadLaunchFile(jsonPath: string): jsonDataType|undefined {
+
   // this function takes the path to a launch.json
   // and returns the contents, or an empty list of configurations
   // if we cannot read the file
-  let existingJSON: any;
-  try {
-    existingJSON = JSON.parse(fs.readFileSync(jsonPath));
-  } catch {
-    // if file is empty ...
-    existingJSON = { configurations: [] };
+  let returnValue:jsonDataType|undefined = undefined;
+  
+  // Requires json-c parsing to handle comments etc.
+  const existingContents = fs.readFileSync (jsonPath).toString();
+  var parseErrors: jsonc.ParseError[] = [];  // not using programatically, for debug only
+  // note that jsonc.parse returns "real json" without the comments
+  const existingJSONdata = jsonc.parse(existingContents, parseErrors, jsoncParseOptions);
+  
+  if (existingJSONdata) {
+    returnValue = { jsonData:existingJSONdata , jsonDataAsString: existingContents };
   }
-  return existingJSON;
+  return returnValue;
 }
 
 export function addLaunchConfiguration(fileUri: Uri) {
   // This function adds the VectorCAST Harness Debug configuration to any
   // launch.json file that the user right clicks on
 
   const jsonPath = fileUri.fsPath;
-  const existingJSON: any = loadLaunchFile(jsonPath);
-
-  // Remember that the vectorJSON data has the "configurations" level which is an array
+  const existingLaunchData:jsonDataType|undefined = loadLaunchFile(jsonPath);
+  
   const vectorJSON = JSON.parse(
     fs.readFileSync(
       path.join(globalPathToSupportFiles, "vcastLaunchTemplate.json")
     )
   );
 
-  const vectorConfiguration = vectorJSON.configurations[0];
+  // if we have a well formatted launch file with an array of configurations ...
+  if (existingLaunchData && existingLaunchData.jsonData.configurations && existingLaunchData.jsonData.configurations.length > 0) {
 
-  // now loop through launch.json to make sure it does not already have the vector config
-  let existingConfigFound = false;
+      // Remember that the vectorJSON data has the "configurations" level which is an array
+    const vectorConfiguration = vectorJSON.configurations[0];
```

Comment:
This surprises me; so the JSON object returned by json-c parser is of a different format than the normal parser?

What happens if there are no configurations in the fie?

Reply:
No the same format.  I just made the vector one a "complete" json file with the configuration level so that I can use it at line 247

Reply:
Sorry, this is my bad; I thought this was `existingLaunchData`.